docs/rfcs/blocks-in-widget-areas.md

```diff
@@ -0,0 +1,535 @@
+
+```js
+function WidgetArea( { id } ) {
+	return (
+		<BlockEditor
+			// Illustrative only: apiGet() is not a real function. In production,
+			// @wordpress/core-data would be used to interface with the REST API.
+			blocks={ apiGet( `/wp/v2/widget-areas/${ id }` ) }
+			onChange={ ( blocks ) => {
+				// Illustrative only: apiPut() is not a real function. In production,
+				// @wordpress/core-data would be used to interface with the REST API.
+				apiPut( `/wp/v2/widget-areas/${ id }`, blocks );
+			} }
+		/>
+	);
+}
+```
+
+[block-editor]: https://github.com/WordPress/gutenberg/tree/master/packages/block-editor
+
+## REST API
+
+Fetching and updating the blocks in a widget area is done via new REST API endpoints.
+
+### The widget area resource
+
+The REST API is built around a single resource object: the widget area.
+
+```json
+{
+	"id": "footer",
+	"name": "Footer",
+	"description": "",
+	"content": {
+		"raw": "
+			<!-- wp:paragraph -->
+			<p>Hello there!</p>
+			<!-- /wp:paragraph -->
+
+			<!-- wp:legacy-widget {\"identifier\":\"search-2\",\"instance\":{\"title\":\"\"}} /-->
+		",
+		"rendered": "
+			<p>Hello there!</p>
+			<form role=\"search\" method=\"get\" class=\"search-form\" action=\"http://wordpress.test/\">
+				...
+			</form>
+		",
+	},
+}
+```
+
+#### `id`
+
+- Type: `string`
+- Access: Read-only
+
+```json
+{ "id": "footer" }
+```
+
+The unique identifier for the widget area. This is set by the theme using [`register_sidebar()`][register_sidebar].
+
+[register_sidebar]: https://developer.wordpress.org/reference/functions/register_sidebar/
+
+#### `name`
+
+- Type: `string`
+- Access: Read-only
+
+```json
+{ "name": "Footer" }
+```
+
+The name or title of the widget area displayed in the admin interface. This is set by the theme using [`register_sidebar()`][register_sidebar].
+
+#### `description`
+
+- Type: `string`
+- Access: Read-only
+
+```json
+{ "description": "" }
+```
+
+Description of the widget area displayed in the admin interface. This is set by the theme using [`register_sidebar()`][register_sidebar]. 
+
+#### `content.raw`
+
+- Type: `string`
+- Access: Read & write
+
+```json
+{ "content": { "raw": "<!-- wp:paragraph --><p>Hello there!</p><!-- /wp:paragraph -->" } }
+```
+
+The raw HTML content of the widget area. Contains Gutenberg block markup and is suitable for loading into the block editor.
+
+#### `content.rendered`
+
+- Type: `string`
+- Access: Read-only
+
+```json
+{ "content": { "rendered": "<p>Hello there!</p>" } }
+```
+
+The rendered HTML content of the widget area. Contains only rendered block markup and is suitable for inserting into a website frontend. Is equivelant to the HTML rendered by the [`dynamic_sidebar()`][dynamic_sidebar] function.
+
+[dynamic_sidebar]: https://developer.wordpress.org/reference/functions/dynamic_sidebar/
+
+### Fetching all widget areas
+
+Fetches all registered widget areas.
+
+#### Request
+
+- Method: `GET`
+- Route: `/wp/v2/widget-areas`
```

Comment:
Again I ask the question, why is this needed, if it is a different post type, then the standard post controller can get a list of widget areas here.

Reply:
Let's keep discussion in the other thread: https://github.com/WordPress/gutenberg/pull/14812/files#r272127421

Reply:
@noisysocks No, let's keep it here, keep everything together, as it is confusing for others, coming to this later.

Reply:
The link above is to a thread in this RFC 🙂